### ISSUE TITLE
Permission denied after update unison and docker-sync

### ISSUE THREAD

**Issue by cduran433** [EugenMayer/docker-sync#548]
I just update unison and docker-sync and after that, I try to up my environment but I find these errors. It is a permission denied when it is trying to sync the folder. 

I make a fresh install but it is still not working. 

Do you have any sugestion?

```
Failed: Error in processing a transfer instruction:
Permission denied [open(/app_sync/.unison.composer.phar.7ab20fb7700158d5547e0bd49f0dd17b.unison.tmp
```

### Error/Feature Requestion/Docs
Error

### Sync strategy
unison

### your docker-sync.yml

```
version: "2"

options:
  compose-file-path: 'docker-compose.yml'
  compose-dev-file-path: 'docker-compose.override.yml'
  verbose: true
syncs:
  #IMPORTANT: ensure this name is unique and does not match your other container names
  docker-mac-symfony-sync:
    src: './symfony'
    sync_strategy: 'unison'
    sync_args:
      - "-prefer newer"
      - "-ignore='Path .git'"
      - "-ignore='BelowPath .git'"
    sync_userid: '1000'
    sync_excludes: ['.gitignore', '.git/', 'app/cache/*']
    # sync_excludes: ['.gitignore', '.git/']
  docker-mac-mysql:
    src: './mysql/db'
    sync_strategy: 'unison'

```

```
services:
    symfony:
        build: ./symfony_container
        links:
            - db
            - mongodb
            - redis
        volumes:    
            - docker-mac-symfony-sync:/var/www/symfony:nocopy

    db:
        image: mysql:5.5
        volumes:    
            - docker-mac-mysql:/var/lib/mysql:nocopy


volumes:
  docker-mac-symfony-sync:
    external: true
  docker-mac-mysql:
    external: true
```

### OS
OSX

**Comment by punkstar** [comment]
I've been seeing this issue too - still in the process of diagnosing the issue but the root cause appears to be that the `/app_sync` directory is not owned by `$OWNER_UID`. It seems like the ownership issue is typically resolved when `precopy_appsync` is executed (https://github.com/EugenMayer/docker-image-unison/blob/f1cca4da04da0e4dbef9305a0d80c6c0a86d0973/precopy_appsync.sh#L13), but the precopy doesn't seem to be executed when I do a `docker-sync start`.

**Comment by punkstar** [comment]
FWIW I can recreate this issue with a more basic setup:

**docker-sync.yml**
```
version: "2"

options:
  composer-file-path: docker-compose.yml
  verbose: true

syncs:
  example-unision:
    sync_strategy: unison
    src: 'src'
    sync_userid: 33
```

**docker-compose.yml**
```
version: '2.0'

services:
  cli: 
    image: php:7.2-cli
    volumes:
      - example-unison:/example/unison:nocopy

volumes:
  example-unison:
    external: true
```

**src/example-content.txt**
```
Hello World!
```

Then, when I run `docker-sync start`, I get the following output:

```
          ok  Starting unison for sync example-unision
          ok  example-unision container not running
          ok  creating example-unision container
     command  docker run -p '192.168.64.3::5000' -v example-unision:/app_sync -e APP_VOLUME=/app_sync -e TZ=${TZ-`readlink /etc/localtime | sed -e 's,/usr/share/zoneinfo/,,'`} -e UNISON_SRC="-socket 5000" -e UNISON_DEST="/app_sync" -e MONIT_ENABLE="false" -e MONIT_INTERVAL="" -e MONIT_HIGH_CPU_CYCLES="" -e UNISON_ARGS="" -e UNISON_WATCH_ARGS="" -e OWNER_UID="33" -e HOSTSYNC_ENABLE="0" -e UNISONSOCKET_ENABLE="1"  --name example-unision -d eugenmayer/unison:2.51.2.0
          ok  starting initial sync of example-unision
     command  docker inspect --format='{{(index (index .NetworkSettings.Ports "5000/tcp") 0).HostPort}}' example-unision
     command  unison -testserver /app_sync "socket://192.168.64.3:32773"
     command  docker inspect --format='{{(index (index .NetworkSettings.Ports "5000/tcp") 0).HostPort}}' example-unision
     command  unison '/Users/nrj/Projects/docker-sync-example/src' -auto -batch -prefer '/Users/nrj/Projects/docker-sync-example/src' -copyonconflict socket://192.168.64.3:32773
       error  Error starting sync, exit code 0
     message  Warning: No archive files were found for these roots, whose canonical names are:
	/Users/nrj/Projects/docker-sync-example/src
	//58098141bbfd//app_sync
This can happen either
because this is the first time you have synchronized these roots,
or because you have upgraded Unison to a new version with a different
archive format.

Update detection may take a while on this run if the replicas are
large.

Unison will assume that the 'last synchronized state' of both replicas
was completely empty.  This means that any files that are different
will be reported as conflicts, and any files that exist only on one
replica will be judged as new and propagated to the other replica.
If the two replicas are identical, then no changes will be reported.

If you see this message repeatedly, it may be because one of your machines
is getting its address from DHCP, which is causing its host name to change
between synchronizations.  See the documentation for the UNISONLOCALHOSTNAME
environment variable for advice on how to correct this.

Donations to the Unison project are gratefully accepted:
http://www.cis.upenn.edu/~bcpierce/unison

file     ---->            example-content.txt
100%  00:00 ETAFailed [example-content.txt]: Error in processing a transfer instruction:
Permission denied [open(/app_sync/.unison.example-content.txt.c80bb7d3d5f71df82f03c903b61cf417.unison.tmp)]

     message  Unison 2.51.2 (ocaml 4.06.1): Contacting server...
Connected [//58098141bbfd//app_sync -> //sterling.local//Users/nrj/Projects/docker-sync-example/src]
Looking for changes
  Waiting for changes from server
Reconciling changes
Propagating updates
UNISON 2.51.2 (OCAML 4.06.1) started propagating changes at 10:08:56.62 on 16 Mar 2018
[BGN] Copying example-content.txt from /Users/nrj/Projects/docker-sync-example/src to //58098141bbfd//app_sync
Failed: Error in processing a transfer instruction:
Permission denied [open(/app_sync/.unison.example-content.txt.c80bb7d3d5f71df82f03c903b61cf417.unison.tmp)]
UNISON 2.51.2 (OCAML 4.06.1) finished propagating changes at 10:08:56.62 on 16 Mar 2018
Saving synchronizer state
Synchronization incomplete at 10:08:56  (0 items transferred, 0 skipped, 1 failed)
  failed: example-content.txt
     success  Unison server started
     command  docker inspect --format='{{(index (index .NetworkSettings.Ports "5000/tcp") 0).HostPort}}' example-unision
     command  unison '/Users/nrj/Projects/docker-sync-example/src' -auto -batch -prefer '/Users/nrj/Projects/docker-sync-example/src' -copyonconflict socket://192.168.64.3:32773
       error  Error starting sync, exit code 0
     message  new file ---->            example-content.txt
100%  00:00 ETAFailed [example-content.txt]: Error in processing a transfer instruction:
Permission denied [open(/app_sync/.unison.example-content.txt.c80bb7d3d5f71df82f03c903b61cf417.unison.tmp)]

     message  Unison 2.51.2 (ocaml 4.06.1): Contacting server...
Connected [//58098141bbfd//app_sync -> //sterling.local//Users/nrj/Projects/docker-sync-example/src]
Looking for changes
  Waiting for changes from server
Reconciling changes
Propagating updates
UNISON 2.51.2 (OCAML 4.06.1) started propagating changes at 10:08:56.72 on 16 Mar 2018
[BGN] Copying example-content.txt from /Users/nrj/Projects/docker-sync-example/src to //58098141bbfd//app_sync
Failed: Error in processing a transfer instruction:
Permission denied [open(/app_sync/.unison.example-content.txt.c80bb7d3d5f71df82f03c903b61cf417.unison.tmp)]
UNISON 2.51.2 (OCAML 4.06.1) finished propagating changes at 10:08:56.72 on 16 Mar 2018
Saving synchronizer state
Synchronization incomplete at 10:08:56  (0 items transferred, 0 skipped, 1 failed)
  failed: example-content.txt
     success  Starting Docker-Sync in the background
```

Some details on my macOS 10.13.3 environment:

```
$ docker-sync -v
0.5.5
$ unison -version
unison version 2.51.2 (ocaml 4.06.1)
$ docker -v
Docker version 18.01.0-ce, build 03596f5
$ docker-compose -v
docker-compose version 1.18.0, build unknown
```

**Comment by sav-valerio** [comment]
Same here on Windows.
```message  Unison 2.51.2 (ocaml 4.06.0): Contacting server...
Connected [//PC/c/Users/user/git -> //d3ffc206adf1//app_sync]
Looking for changes
  Waiting for changes from server
Reconciling changes
Propagating updates
UNISON 2.51.2 (OCAML 4.06.0) started propagating changes at 17:56:27.77 on 16 Mar 2018
[BGN] Copying properties for  from /c/Users/user/git to //d3ffc206adf1//app_sync
[BGN] Copying redacted1 from /c/Users/user/git to //d3ffc206adf1//app_sync
[BGN] Copying redacted2 from /c/Users/user/git to //d3ffc206adf1//app_sync
[BGN] Copying redacted3 from /c/Users/user/git to //d3ffc206adf1//app_sync
[BGN] Copying redacted4 from /c/Users/user/git to //d3ffc206adf1//app_sync
Failed: Error in setting permissions:
Operation not permitted [chmod(/app_sync)]
Failed: Error in creating directory:
Permission denied [mkdir(/app_sync/.unison.redacted1.0210a7ef71983c17e7763087129e1209.unison.tmp)]
Failed: Error in creating directory:
Permission denied [mkdir(/app_sync/.unison.redacted2.0210a7ef71983c17e7763087129e1209.unison.tmp)]
Failed: Error in creating directory:
Permission denied [mkdir(/app_sync/.unison.redacted3.0210a7ef71983c17e7763087129e1209.unison.tmp)]
Failed: Error in creating directory:
Permission denied [mkdir(/app_sync/.unison.redacted4.0210a7ef71983c17e7763087129e1209.unison.tmp)]
UNISON 2.51.2 (OCAML 4.06.0) finished propagating changes at 17:56:27.82 on 16 Mar 2018
Saving synchronizer state
Synchronization incomplete at 17:56:27  (0 items transferred, 0 skipped, 5 failed)````

**Comment by EugenMayer** [comment]
fixed with 0.5.6
